### PR TITLE
Update instruction for using Tilt environment

### PR DESCRIPTION
**Pull request by Xenwar**
Update instruction for using Tilt environment

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Since prow is down, there is no tide, but this `lgtm`.

/test-integration

**Comment by fmuyassarov** [comment]
@Xenwar would you mind to rename the title to describe the actual bug/fix in short?

**Comment by Xenwar** [comment]
@kashifest @fmuyassarov Instruction updated according to the discussion we had. please review

File: docs/dev-setup.md [hunk 1]
```diff
@@ -123,13 +123,33 @@ After you have cloned both repositories, your folder structure should look like:
...
+```bash
+cluster-api$ make kustomize
+hack/ensure-kustomize.sh
+fetching kustomize@3.9.1
+
+cluster-api$ ./hack/tools/bin/kustomize version
+{Version:kustomize/v3.9.1 GitCommit:7439f1809e5ccd4677ed52be7f98f2ad75122a93 BuildDate:2020-12-29T19:25:35Z GoOs:linux GoArch:amd64}
+```
+
+If the abov installation does not work, install kustomize using a package manager and move the binary to ```hack/tools/bin/kustomize version``` in the cluster-api repo.
```
Comment (fmuyassarov) [inline docs/dev-setup.md:143]:
```suggestion
If the above installation does not work, install kustomize using a package manager and move the binary to ```hack/tools/bin/kustomize version``` in the cluster-api repo.
```

File: docs/dev-setup.md [hunk 2]
```diff
@@ -123,13 +123,33 @@ After you have cloned both repositories, your folder structure should look like:
...
+```bash
+git checkout release-0.3
+```
+
+Install kustomize
+
+```bash
+cluster-api$ make kustomize
+hack/ensure-kustomize.sh
+fetching kustomize@3.9.1
```
Comment (fmuyassarov) [inline docs/dev-setup.md:137]: Nit: maybe ?
```suggestion
```bash
pushd cluster-api
make kustomize
hack/ensure-kustomize.sh
```

File: docs/dev-setup.md (hunk 1 above)
Comment (furkatgofurov7) [inline docs/dev-setup.md:143]: The path to move the binary seems to be incorrect to me. Do we need `version`?

**Review by furkatgofurov7 — COMMENTED** [review]
`lgtm` cancel since there were additional changes.

File: docs/dev-setup.md (hunk 1 above)
Comment (Xenwar) [reply docs/dev-setup.md:143]: fixed

Comment (Xenwar) [reply docs/dev-setup.md:143]: fixed

File: docs/dev-setup.md (hunk 2 above)
Comment (Xenwar) [reply docs/dev-setup.md]: There was a mix up of the commands and their output. 
removed the outputs

File: docs/dev-setup.md [hunk 3]
```diff
@@ -123,13 +123,31 @@ After you have cloned both repositories, your folder structure should look like:
...
+If the above installation does not work, install kustomize using a package manager and move the binary to ```hack/tools/bin/kustomize``` in the cluster-api repo.
+
 After configuring the environment variables, run the following to generate your
 `tilt-settings.json` file in the cluster API repository:
 
 ```shell
 cat <<EOF > tilt-settings.json
 {
     "allowed_contexts": ["kind-capm3"],
+    "default_registry": "gcr.io/cluster-api-provider",
```
Comment (kashifest) [inline docs/dev-setup.md:150]: Why is this needed @Xenwar ?

Comment (Xenwar) [reply docs/dev-setup.md:150]: the command that uses tis configuration file requires it. It was needed before

**Review by fmuyassarov — COMMENTED** [review]
/lgtm
